core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -474,13 +477,13 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
    * Spark fair scheduler pool.
    */
   def setLocalProperty(key: String, value: String) {
-    if (localProperties.get() == null) {
-      localProperties.set(new Properties())
+    if (localProperties.get().isEmpty) {
+      localProperties.set(Some(new Properties()))
```

Comment:
@JoshRosen the idea is to use Option so compiler can help us prevent NullPointerException.  And by eagerly setting localProperties to None, localProperties won't be null.  As a contributor, when I see the type is Option[Properties], I know I'm safe from NPE, because I might not aware of the eagerly setting to empty Properties somewhere in the codes, and end up putting all ```if (localProperties.get != null)``` code everywhere.  If we can change DeveloperApi, I would like to change all ```properties = null``` to Option.

That's just my 2 cents, I'm more than happy to discuss and listen to your thoughts.